tests/v2/validation/rbac/psa/psa.go

```diff
@@ -270,3 +147,40 @@ func editPsactCluster(client *rancher.Client, clustername string, namespace stri
 	}
 	return clusterType, nil
 }
+
+func getAndConvertNamespace(namespace *v1.SteveAPIObject, steveAdminClient *v1.Client) (*coreV1.Namespace, error) {
+	getNSSteveObject, err := steveAdminClient.SteveType(namespaces.NamespaceSteveType).ByID(namespace.ID)
+	if err != nil {
+		return nil, err
+	}
+	namespaceObj := &coreV1.Namespace{}
+	err = v1.ConvertToK8sType(getNSSteveObject.JSONResp, namespaceObj)
+	if err != nil {
+		return nil, err
+	}
+	return namespaceObj, nil
+}
+
+func createRole(client *rancher.Client, context string, roleName string, rules []management.PolicyRule) (role *management.RoleTemplate, err error) {
```

Comment:
is there no v1 object for this? we should try not to add v3 stuff anymore to the tests/code.

Reply:
these were not new tests but already existing ones. In this PR, I only refactored the structure. we can add tests and update these in a new PR.